core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -168,6 +168,23 @@ case class ExecutorLostFailure(execId: String) extends TaskFailedReason {
 
 /**
  * :: DeveloperApi ::
+ * The task failed because the executor that it was running on was prematurely terminated. The
+ * executor is forcibly exited but the exit should be considered as part of normal cluster
+ * behavior.
+ */
+@DeveloperApi
+case class ExecutorForTaskExited(
+    taskId: Long,
+    execId: String,
+    exitReason: String,
+    exitCode: Int)
+    extends TaskFailedReason {
+  override def toErrorString: String = s"Task with ID $taskId had its executor ${execId}" +
```

Comment:
it's probably OK to leave out the `taskId` here since this is displayed on the UI next to the task ID anyway. Then you can remove the field from this case class.